File: 1052/solution.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int curr = 0;
        for (int i = 0; i < customers.length; i++) {
            curr += grumpy[i] == 1 ? 0 : customers[i];
        }
        int res = curr;
        for (int i = 0; i < customers.length; i++) {
            curr += grumpy[i] == 1 ? customers[i] : 0;
            if (i >= minutes) {
                curr -= grumpy[i - minutes] == 1 ? customers[i - minutes] : 0;
            }
            res = Math.max(res, curr);
        }
        return res;
    }
}

// time O(n)
// space O(1)

File: 1052/solution.py
class Solution:
    def maxSatisfied(
        self, customers: list[int], grumpy: list[int], minutes: int
    ) -> int:
        curr = 0
        for i in range(len(customers)):
            curr += customers[i] if not grumpy[i] else 0
        res = curr
        for i in range(len(customers)):
            curr += customers[i] if grumpy[i] else 0
            if i >= minutes:
                curr -= customers[i - minutes] if grumpy[i - minutes] else 0
            res = max(res, curr)
        return res
